src/ds/champ_map.h

```diff
@@ -143,8 +145,8 @@ namespace champ
         const auto& entry = bin[i];
         if (k == entry->key)
         {
-          const auto diff =
-            map::get_size(entry->key) + map::get_size(entry->value);
+          const auto diff = map::get_serialized_size_with_padding(entry->key) +
+            map::get_serialized_size_with_padding(entry->value);
```

Comment:
Slightly less buggy with `remove()`: we did not return the **padded** size.

Whether padding is still necessary in snapshots is left for future work (https://github.com/microsoft/CCF/issues/4731).